source/example1_sync_fifo.rst

```diff
@@ -1,2 +1,103 @@
 Sync FIFO
-============
\ No newline at end of file
+============
+.. code-block:: systemverilog
+
+    `timescale 1ns / 1ps
+    //////////////////////////////////////////////////////////////////////////////////
+    // Company: URBADASS.LTD
+    // Engineer: zick
+    // 
+    // Create Date: 02/26/2023 09:01:54 PM
+    // Design Name: sync fifo handshake
+    // Module Name: sync_fifo
+    // Project Name: sync fifo
+    // Description: 
+    //              a handshake based sync fifo for serialized data
+    // Dependencies: 
+    // 
+    // Revision:
+    //              Revision 0.01 - File Created
+    // Additional Comments:
+    // 
+    //////////////////////////////////////////////////////////////////////////////////
+    module sync_fifo
+        #(
+            parameter int unsigned DEPTH    = 8             ,
```

Comment:
It looks like DEPTH can only be the Nth power of 2? If so, we need more comments to clarify this thing, or use other input like POINTER_WIDTH, or modify the design to make it more flexible.